### ISSUE TITLE
XEH initPost runs twice when added retroactively via function

### ISSUE THREAD

**Issue by SilentSpike** [CBATeam/CBA_A3#567]
**Arma 3 Version:** `1.66` (stable)
**CBA Version:** `3.1.2` (dev 311d8d7128f4f39679e984164736f34f9d42c6f3)

**Mods:**
- `\x\cba`
- `\z\ace`
- `@devcon_a3` (in-house dev console that I always run while developing)

**Description:**
- When using function `CBA_fnc_addClassEventHandler` to add an `"initPost"` XEH to a vehicle with the retroactive parameter set to true, the event will run twice for existing instances of that vehicle.

**Steps to reproduce:**
Run the following repro mission:
[EmptyMission.VR.zip](https://github.com/CBATeam/CBA_A3/files/666150/EmptyMission.VR.zip)

You can see I've added an initPost EH to `CAManBase` (player unit) both via the `CBA_fnc_addClassEventHandler` function and via the `Extended_InitPost_EventHandlers` in description.ext

The config version runs only once, while the scripted version runs twice.

**Where did the issue occur?**
- Editor (Singleplayer)

**RPT log file:**
- [Arma3_2016-12-21_11-23-18.txt](https://github.com/CBATeam/CBA_A3/files/666195/Arma3_2016-12-21_11-23-18.txt)

**Comment by commy2** [comment]
Does this always happen regardless of when the eventhandler is used? preInit vs. postInit vs. mid mission / scheduled init.sqf?

**Comment by SilentSpike** [comment]
Currently only tested from postInit, will try init.sqf and preInit also

**Comment by SilentSpike** [comment]
It only runs once from both preInit and init.sqf, so it seems to only be an issue when added during postInit

**Comment by commy2** [comment]
Very interesting. That narrows down the problem alot.

**Comment by commy2** [comment]
hijacking this for a note:
- decouple CBA postInit from BI postInit to avoid getting stalled with `waitUntil`

**Comment by PabstMirror** [comment]
`(entities [[], [], true, true])` returns duplicates of units in vehicles when called at postInit

AI_2 starts in a car, listed twice
`[L Modules:1,AI_2,aCar,startingPlayer,AI_1,AI_2]`

**Comment by commy2** [comment]
So that is all? That fixes this issue? Dammit BI.

**Comment by Dystopian** [comment]
looks like this is not fixed
> 2:13:47 "Scripted initPost ran"
 2:13:47 "Config initPost ran"
 2:13:47 "Scripted initPost ran"

**Comment by commy2** [comment]
Repro?

**Comment by Dystopian** [comment]
[EmptyMission.VR.zip](https://github.com/CBATeam/CBA_A3/files/1234755/EmptyMission.VR.zip)
this is simplified @SilentSpike mission, it can be tested with CBA only
